tools/shared/src/main/scala/org/scalajs/core/tools/linker/checker/IRChecker.scala

```diff
@@ -203,7 +212,7 @@ private final class IRChecker(unit: LinkingUnit, logger: Logger) {
     val FieldDef(static, name, tpe, mutable) = fieldDef
     implicit val ctx = ErrorContext(fieldDef)
 
-    if (static && !classDef.kind.isJSClass)
+    if (static && classDef.kind.isJSType && !classDef.kind.isJSClass)
       reportError(s"FieldDef '$name' cannot be static")
```

Comment:
IIUC, this check is bogus. The only things it triggers on are `AbstractJSType | NativeJSClass | NativeJSModuleClass` and these do not allow fields on a much higher level.